### PR TITLE
Waterfall initial

### PR DESCRIPTION
**Pull request by dmt0**

### CODE REVIEW

**Comment by dmt0** [comment]
@nicolaskruchten please take a look at naming and positioning of things. Note, a few things don't show up, I've raised a few questions with plotly.js team.

**Comment by dmt0** [comment]
f35bf0b is mainly for clients who use trace type selector and add more trace types to it. We want the modal to stay same shape.

**Comment by nicolaskruchten** [comment]
* let's remove inside/outside text font stuff... we already allow overall specification
* can you bring in one of the waterfall mocks from the PR? or update the mock list in the dev dropdown?
* percy seems to show the scatter screenshot for style/traces for some reason
* please add the 'measure' dropdown to the structure/traces panel, as this one is super important

**Comment by nicolaskruchten** [comment]
also let's merge this one once we have the icons in the upstream repo otherwise we can't deploy

**Comment by dmt0** [comment]
Looks like we can't have a percy mock of waterfall, because of that plotly-cartesian issue.

**Comment by nicolaskruchten** [comment]
Remind me what the Cartesian issue is?

**Comment by dmt0** [comment]
Actually, it's quite solvable, I'm gonna take a stab at it. 
Test fixture imports plotly-cartesian, because with full plotly there are issues mocking out some browser functions (don't remember the whole story). But plotly-cartesian doesn't have all plots.
Storybook/percy works with full plotly, Jest tests work only with plotly-cartesian.
So I'm just gonna pass the right plotly to the test fixture...

**Comment by dmt0** [comment]
Managed to cut 1 minutes off of CI build time as well 🎉 
And we uncovered something interesting in percy that we didn't see before.
And now we can add all kinds of other mocks.

File: dev/mocks.json [hunk 1]
```diff
@@ -42,6 +43,9 @@
   "animation.json",
   "annotations-autorange.json",
   "annotations.json",
+  "automargin-push-x-extra.json",
```
Comment (VeraZab) [inline dev/mocks.json]: where are all these from?

File: src/components/fields/TextPosition.js [hunk 2]
```diff
@@ -91,7 +91,11 @@ export default connectToContainer(UnconnectedTextPosition, {
       {label: _('Bottom Center'), value: 'bottom center'},
       {label: _('Bottom Right'), value: 'bottom right'},
     ];
-    if (context.container.type === 'pie' || context.container.type === 'bar') {
+    if (
+      context.container.type === 'pie' ||
```
Comment (VeraZab) [inline src/components/fields/TextPosition.js]: ['pie', 'bar', 'waterfall'].includes(context.container.type) ?

File: src/default_panels/GraphCreatePanel.js [hunk 3]
```diff
@@ -57,6 +57,12 @@ const GraphCreatePanel = (props, {localize: _, setPanel}) => {
         }}
         attr="z"
       />
+      <DataSelector
+        label={{
+          '*': _('Measure'),
```
Comment (VeraZab) [inline src/default_panels/GraphCreatePanel.js]: I don't really understand that key there: `*`
https://plotly.github.io/schema-viewer/?attribute=measure

could you point me to the reference that explains why its `*`?

File: src/default_panels/StyleTracesPanel.js [hunk 4]
```diff
@@ -78,6 +78,17 @@ const StyleTracesPanel = (props, {localize: _}) => (
       <Numeric label={_('Max Tube segments')} attr="maxdisplayed" />
     </PlotlySection>
     <MultiColorPicker label={_('Color')} attr="color" />
+    <TraceTypeSection name={_('Waterfall')} traceTypes={['waterfall']} mode="trace">
+      <LayoutSection attr="name">
+        <Radio
+          label={_('Mode')}
+          attr="waterfallmode"
+          options={[{label: _('Group'), value: 'group'}, {label: _('Overal'), value: 'overlay'}]}
```
Comment (VeraZab) [inline src/default_panels/StyleTracesPanel.js]: lol: https://plotly.github.io/schema-viewer/?attribute=waterfallmode
Label: 'Overlay' : )

**Comment by VeraZab** [comment]
codewise, small comments, the only one to fix really is the lable for Overlay, otherwise 💃

File: dev/mocks.json (hunk 1 above)
Comment (dmt0) [reply dev/mocks.json]: plotly.js repo

File: src/default_panels/StyleTracesPanel.js (hunk 4 above)
Comment (dmt0) [reply src/default_panels/StyleTracesPanel.js]: wow, how did that get in there!

File: dev/mocks.json (hunk 1 above)
Comment (dmt0) [reply dev/mocks.json]: Maybe sometime I'll set up a thing to pull the list automatically, we really don't need to keep this file in the archive - it's just a directory listing essentially.
https://github.com/plotly/plotly.js/tree/master/test/image/mocks

File: src/default_panels/GraphCreatePanel.js (hunk 3 above)
Comment (dmt0) [reply src/default_panels/GraphCreatePanel.js]: It's a way to give it different names depending on trace type - it's just a construct in `DataSelector`

File: src/default_panels/StyleTracesPanel.js [hunk 5]
```diff
@@ -564,6 +599,29 @@ const StyleTracesPanel = (props, {localize: _}) => (
         options={[{label: _('Show'), value: true}, {label: _('Hide'), value: false}]}
       />
     </PlotlySection>
+    <PlotlySection name={_('Totals Styles')}>
```
Comment (nicolaskruchten) [inline src/default_panels/StyleTracesPanel.js]: can you rename this to "Total Marker Styles" and the other two from "X Trace Styles" to "X Marker Styles" plz?

**Comment by nicolaskruchten** [comment]
✨ once you've renamed the one thing above :)

**Comment by nicolaskruchten** [comment]
nice job on fixing Percy :)

**Pull request merged**